File: android/app/src/main/java/edu/gatech/cs2340/rajsjarfiles/spacetrader/entity/player/ShipType.java
package edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Represents a ship type.
 */
public enum ShipType implements Serializable {
    FLEA(25, 0),
    GNAT(100, 1),
    FIREFLY(100, 1),
    MOSQUITO(100, 2),
    BUMBLEBEE(100, 1),
    BEETLE(50, 0),
    HORNET(150, 3),
    GRASSHOPPER(150, 2),
    TERMITE(200, 1),
    WASP(200, 3);

    private int maxHealth;

    private int maxWeaponSlots;

    /**
     * Constructor for a ship type.
     *
     * @param maxHealth the max health of the ship
     * @param maxWeaponSlots the max weapons slots the ship has
     */
    ShipType(int maxHealth, int maxWeaponSlots) {
        this.maxHealth = maxHealth;
        this.maxWeaponSlots = maxWeaponSlots;
    }

    /**
     * @return the max health of the ship
     */
    public int getMaxHealth() {
        return maxHealth;
    }

    /**
     * @return the max weapons slots the ship has
     */
    public int getMaxWeaponSlots() {
        return maxWeaponSlots;
    }

    /**
     * @return ship types that can have weapons
     */
    public static List<ShipType> getShipsWithWeapons() {
        List<ShipType> list = new ArrayList<>();
        for (ShipType s : ShipType.values()) {
            if (s.maxWeaponSlots > 0) {
                list.add(s);
            }
        }
        return list;
    }
}

File: ios/SpaceTrader/SpaceTrader/src/entity/player/ShipType.swift
//
//  .swift
//  SpaceTrader
//
//  Created by Thomas Suarez on 4/20/19.
//  Copyright © 2019 RajsJarFiles. All rights reserved.
//

import Foundation

struct ShipType {
    public static let Flea = ShipType(25, 0)
    public static let Gnat = ShipType(100, 1)
    public static let Firefly = ShipType(100, 1)
    public static let Mosquito = ShipType(100, 2)
    public static let Bumblebee = ShipType(100, 1)
    public static let Beetle = ShipType(50, 0)
    public static let Hornet = ShipType(150, 3)
    public static let Grasshopper = ShipType(150, 2)
    public static let Termite = ShipType(200, 1)
    public static let Wasp = ShipType(200, 3)
    
    var types = [
        Flea,
        Gnat,
        Firefly,
        Mosquito,
        Bumblebee,
        Beetle,
        Hornet,
        Grasshopper,
        Termite,
        Wasp
    ]
    
    var maxHealth : Int
    var maxWeaponSlots : Int
    
    init(_ maxHealth : Int, _ maxWeaponSlots : Int) {
        self.maxHealth = maxHealth
        self.maxWeaponSlots = maxWeaponSlots
    }
    
    func getShipsWithWeapons() -> [ShipType] {
        var ships : [ShipType] = []
        for s in types {
            if s.maxWeaponSlots > 0 {
                ships.append(s)
            }
        }
        return ships
    }
}
